x-pack/agent/pkg/agent/application/info/agent_id.go

```diff
@@ -80,6 +80,10 @@ func getInfoFromStore(s ioStore) (*persistentAgentInfo, error) {
 			errors.M(errors.MetaKeyPath, AgentConfigFile))
 	}
 
+	if err := reader.Close(); err != nil {
```

Comment:
How is this related to this PR?

Reply:
this fixes an issue on windows i experienced on tests by introducing more extensive use of agentInfo.ID()